### ISSUE TITLE
[Bug] Some videos only have audio when they first play, are muted after looping

### ISSUE THREAD

**Issue by Sirs0ri** [tuskyapp/Tusky#3145]
I noticed that a small amount of videos lose sound when they start over after having played once. 
Closing out of the media viewer and reopening it makes the sound reappear.

Some videos that show the problem: 
* https://infosec.exchange/@i0null/109496669134948846
* https://infosec.exchange/@i0null/109355838409269982

A video without the problem:
* https://infosec.exchange/@i0null/109638557251991549

I'm not sure if I've seen the issue with accounts other than infosec.exchange/@i0null, but it definitely does not apply to all their videos.
I'm seeing the problem in both the current stable and nightly versions.

* * * *
- Tusky Version: 20.0 and 20.0-0328c4e8
- Android Version: 12
- Android Device: Galaxy s10e
- Mastodon instance (if applicable): n.a.?

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by mcclure** [comment]
I can't reproduce this on Tusky 21 on Android 12 (XPeria 5 III) with these same videos.

**Comment by Sirs0ri** [comment]
I just double-checked, it's still an issue for me with both examples, current nightly (21.0-52c98749) and stable (21.0) builds

If there's any additional info I can provide, please let me know what!

**Comment by nikclayton** [comment]
@Sirs0ri -- what's your device? And does this still occur with Tusky 23?

**Comment by Sirs0ri** [comment]
@nikclayton yes, it's still happening on the current stable (`23.0`) and nightly (`23.0-a7513124`) builds.

I have a Samsung Galaxy S10e, running the latest available official OneUI 4.1 / Android 12.

Note mostly for me: Since the videos linked in the original post have since been removed from our instance's cache, here's a newer one where the problem still occurs: https://infosec.exchange/@i0null/110714603208822014

**Comment by nikclayton** [comment]
Android 12 is API level 31. I've just tried reproducing this in an Android emulator at API 31 and Tusky 23.0, and it doesn't reproduce. The video starts playing, with audio, then loops, and the audio is there in the loop as expected.

@Sirs0ri Are you able to build and install Android apps? The Exoplayer developers have a demo app that can be configured to display different videos. If you can reproduce the problem in their demo app that would help them troubleshoot the problem.

If this is something you can do then let me know, and I'll provide more details on how to do it.

**Comment by Sirs0ri** [comment]
@nikclayton building and testing an app shouldn't be an issue, sure thing!

**Comment by nikclayton** [comment]
Great. So.

1. Fork and clone https://github.com/androidx/media
2. Edit `demos/main/src/main/assets/media.exolist.json`, and add a section like:

```json
  {
    "name": "Mastodon example",
    "samples": [
      {
        "name": "Dinosaurs excerpt",
        "uri": "https://media.infosec.exchange/infosecmediaeu/media_attachments/files/110/714/583/502/451/128/original/064bba7d27663c02.mp4"
      }
    ]
  },
```

(that's the URL to the video from https://infosec.exchange/@i0null/110714603208822014)

3. Edit `demos/main/src/main/java/androidx/media3/demo/main/PlayerActivity.java` and add `player.setRepeatMode(Player.REPEAT_MODE_ALL);` in `initializePlayer()`, just before the call to `playerView.setPlayer(player);`. This is so the video loops to match your problem report.

4. From the root of that project, and with an Android debuggable device connected (ideally your physical device, since that's the one you're seeing problems with), run `.\gradlew :demo:installNoDecoderExtensionsDebug`.

5. Run the newly installed "ExoPlayer" app on your device.

6. Tap the "Mastodon example" section, then tap the video link to start playing it.

If it plays, loops, and the audio remains then, good, that means it'll work on your device when https://github.com/tuskyapp/Tusky/pull/3857 is merged in.

If it doesn't work, file a bug with the media3 folk. Here's a recent bug that I filed for a different video that does fail on my device and in the emulator: https://github.com/androidx/media/issues/524

**Comment by Sirs0ri** [comment]
@nikclayton sorry, took me a second to try this.

With the test player, the video plays and loops correctly!

**Comment by Sirs0ri** [comment]
Fixed in the latest Beta that has #3857 merged! 🎉
